packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -1086,15 +1094,28 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                     if (handlePath.length > 0 && !handlePath.startsWith("/")) {
                         handlePath = `/${handlePath}`;
                     }
-                    id = `${parentHandle}/.app${handlePath}`;
-
-                    // TODO: SPO will deprecate this soon
-                    if (summaryObject.handleType === api.SummaryType.Commit) {
-                        value = {
-                            content: id,
-                        };
+                    const pathKey = `.app${handlePath}`;
+                    const summaryTreeToExpand = this.treesPathToSummaryTreeCache.get(pathKey);
```

Comment:
I just want to better understand what this code does (having comments here would be super useful, as next reader will likely have same questions, and may potentially even regress logic because he/she does not understand it).

This is tree expansion is done such that we can safely detect (and not reference in the future) deleted blobs in current summary, right?
The side-effect of it (if I got it right) are
1. we always send whole tree no matter what, even if some part of the tree did not change.
2. We keep whole tree (including actual blobs) in memory.
3. We calculate SHAs and do deduping for blobs / parts of the tree that did not change.

I think we need to find a way to avoid these side-effects.
Keeping the tree is fine, but every leaf node (blob) needs to be converted from content to reference, i.e. handle + path (plus tracking its SHA separately in SHA -> path map) once summary is uploaded.
That way we only have con # 1, which is not great, but likely is acceptable.

did I get it right?

Reply:
Yes, here we expand the summary tree in order to dedup the blobs that were not deduped in previous summary. This also enables to put the right blobs or delete the wrong blobs from cache. #2 and #3 are not a problem now after having pathToSha cache as don't keep the full blobs in memory anymore. This also prevents to calculate the hash again n again. However we do upload the full tree anymore.